Main/Source/game.cpp

```diff
@@ -3270,50 +3277,93 @@ int game::Load(cfestring& saveName)
   return LOADED;
 }
 
-festring game::SaveName(cfestring& Base)
+/**
+ * this prevents all possibly troublesome characters in all OSs
+ */
+void fixChars(festring& fs)
 {
-  festring SaveName = GetSaveDir();
+  for(festring::sizetype i = 0; i < fs.GetSize(); ++i)
+  {
+    if(fs[i]>='A' && fs[i]<='Z')continue;
+    if(fs[i]>='a' && fs[i]<='z')continue;
+    if(fs[i]>='0' && fs[i]<='9')continue;
+
+    fs[i] = '_';
+  }
+}
+
+bool chkAutoSaveSuffix(festring& fs,bool bAlsoFixIt=false){DBG1(fs.CStr());
+  std::string strChk;
+  strChk = fs.CStr();
+  int i = strChk.find(AUTOSAVE_SUFFIX);
```

Comment:
You can also use festring::FindLast()

Reply:
we actually need a findFirst thing as we can have "PlayerName.AutoSave.AutoSave" use case,
I should have implemented it into festring but I found festring code quite difficult to follow when I thought on doing that in the past, so better not change what I dont understand well :)

Reply:
Then `festring::Find()` and `SEARCH_N_REPLACE` is enough. Also need to check if the final string is empty.

Mixing std::string and festring is a bit funny. :P

Reply:
I was thinking, SearchAndReplace() will work if nothing unexpected happens.
"player_name.AutoSave.hello" will provide unpredictable result
I dont know.. I like to do what is safest (considering funny users typing funny things just to see what happens :)) even if beyond the scope of the function name "chkAutoSaveSuffix()".
Feel free to restrict the scope, the beyond things may never happen anyway :).

Reply:
Need a while loop to fix "..AutoSaveAutoSave.AutoSave"? But "." will be replaced with "_" for save files, so such names will only be detected in two situations:

1. Loading a save file from an unknown version of the game. (with the user's own patch?)
2. The save file was manipulated by the player outside the game.

I think there is no need to fix the name because it is very unlikely that the fixed name will match an existing file.

Reply:
the fix for unprobable situations, despite useful, is not required.

but it is required on this flow:

```
truth game::Init(cfestring& loadBaseName){
  festring absLoadNameOk;
  if(!loadBaseName.IsEmpty()){
    absLoadNameOk = SaveName(loadBaseName,true); //will prepend the path
...
festring game::SaveName(cfestring& Base,bool bLoadingFromAnAutosave){
...
     chkAutoSaveSuffix(CurrentBaseSaveFileName,true);
```

becase the chosen savegame to be loaded may be an .AutoSave, and it is important to keep the CurrentBaseSaveFileName w/o that suffix

Reply:
btw, I opted for using std::string because of it's quite useful `.substr(` method, but indeed I dont expect that after .AutoSave has anything other than another .AutoSave :)
this means, if one day there is something, it will be missing and such code will have to be modified.
but... I suggest we dont put anything after .AutoSave just to not make things more complicated xD

Reply:
Ah, okay. That function name is misleading… It "adjust"s, not "fix"es the filename. Then you should only erase ".AutoSave" at the end of the string, while still searching without any restriction for the return value `true`/`false`. If ".AutoSave" does not end the string (theoretically), then you will get a false-positive result for ".AutoSave fun" on Line 3357.

Reply:
`If ".AutoSave" does not end the string` 
mmm... if .AutoSave is anywhere else than in the end, ABORT(), this may be good! and dont sound that complex

Reply:
Just realized there are also ".AutoSave.sav", ".AutoSave.sav.bkp" and the like. :P

Reply:
and the quite hard to see .AutoSave.sav.tmp (crashing on creating it will keep .AutoSave.sav and the .bkp untouched :))